From a085090ec68efccd4d82c7cf4ea6f18fb475c4d8 Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Thu, 2 Mar 2017 14:36:56 +0800
Subject: [PATCH] gst-player: fix gst-player failed to load external subtitle
 uri

gst_player_set_uri_internal shouldn't free suburi which maybe set
by user to load external subtitle before start play. It just need
reset playbin's subutri property to NULL no matter if there was a
previous one or not.

Upstream-Status: Backport[1.11.3]
https://bugzilla.gnome.org/show_bug.cgi?id=779453

Signed-off-by: Haihua Hu <jared.hu@nxp.com>
---
 gst-libs/gst/player/gstplayer.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/gst-libs/gst/player/gstplayer.c b/gst-libs/gst/player/gstplayer.c
index fc5463a..3757a09 100644
--- a/gst-libs/gst/player/gstplayer.c
+++ b/gst-libs/gst/player/gstplayer.c
@@ -582,12 +582,7 @@ gst_player_set_uri_internal (gpointer user_data)
         (GDestroyNotify) uri_loaded_signal_data_free);
   }
 
-  /* if have suburi from previous playback then free it */
-  if (self->suburi) {
-    g_free (self->suburi);
-    self->suburi = NULL;
-    g_object_set (self->playbin, "suburi", NULL, NULL);
-  }
+  g_object_set (self->playbin, "suburi", NULL, NULL);
 
   g_mutex_unlock (&self->lock);
 
@@ -667,6 +662,9 @@ gst_player_set_property (GObject * object, guint prop_id,
       g_free (self->redirect_uri);
       self->redirect_uri = NULL;
 
+      g_free (self->suburi);
+      self->suburi = NULL;
+
       self->uri = g_value_dup_string (value);
       GST_DEBUG_OBJECT (self, "Set uri=%s", self->uri);
       g_mutex_unlock (&self->lock);
-- 
1.9.1

